README.md

```diff
@@ -1,7 +1,63 @@
+- Install [Yarn](https://yarnpkg.com/lang/en/docs/install/)
+
+### <a name="download">Download the source code</a>
+1. Open up your favorite kind of console
+2. Navigate to the folder in which you want to store the source code
+3. Run `git clone git@github.com:Lapanti/ts-react-boilerplate.git`
+
+### <a name="startingdevelopment">Starting development</a>
+1. Open up the source code in your favorite TypeScript-capable editor (I recommend [Visual Studio Code](https://code.visualstudio.com/) if you don't have a preference)
+2. Run `yarn` in the console to install dependencies (it'll take a while on the first run, so go on and read ahead while you wait)
+3. Read through the comments in all the source files to get yourself acquinted with the ideas, concepts and patterns
+4. Start the application by running `yarn run develop` in your console (inside the folder you downloaded the code to) and open up your browser in the address it prints out
+    - If you want the use of [LiveReload](http://livereload.com/), install a browser extension in [Chrome](https://chrome.google.com/webstore/detail/livereload/jnihajbhpnppcggbcgedagnkighmdlei?hl=en) or [Firefox](https://addons.mozilla.org/en-gb/firefox/addon/livereload/)
+5. Create a deployable version of the application by running `yarn run build`
+6. Start the deployable version by running `yarn run start` or read the [How to Docker](#dockerization) guide to Dockerize your application
+7. To test your application, run `yarn run test`
+8. Start modifying the code to build your own application
+9. ???
+10. Profit (like all those SaaS startups you've heard about)
+
+## <a name="tipsandsuggestions">Tips and suggestions</a>
+- Make sure everything has a type (the more you squeeze out of the compiler the easier you're going to have it while developing)
+- Follow [BEM](http://getbem.com/naming/)-naming with CSS
+- Follow [Redux-ducks pattern](https://github.com/erikras/ducks-modular-redux) except that name the reducers as according to the file (see [IndexReducer.tsx](src/modules/index/IndexReducer.tsx) for an example)
+
+## <a name="dockerization">How to Docker</a>
+The [Dockerfile](Dockerfile) is where you can find the configuration to build a [Docker](https://www.docker.com/) image out of your application. The first line of the `Dockerfile` (starting with `FROM`) includes the base for your Dockerfile, feel free to change it if you want to.
+1. Put your email to the [fourth line in the Dockerfile](Dockerfile#L4)
+2. In your console run `docker build .`
+3. In your console run `docker run -d -p 8888:8888 bd9b1d6725bc` **but** replace `bd9b1d6725bc` with the image ID you received from the previous command
+4. Host your Docker image in your favorite cloud or local server (the web is filled with guides for this)
+
+## <a name="dependencies">Dependencies</a>
+The following are all the dependencies of the project, with the reasoning behind their includement:
```

Comment:
`s/includement/inclusion/` or `s/ing behind their includement/ for including them/`